identity-service/src/routes/cognito.js

```diff
@@ -75,31 +75,58 @@ module.exports = function (app) {
           headers
         })
 
+        // https://cognitohq.com/docs/reference#flow_get_flow_session
+        // user is always present
         const userInfo = flowSessionResponse.data.user
+
+        // id_number is always present but nullable
+        // phone is always present but nullable
+        // date_of_birth is always present but nullable
+        // address is always present but nullable
+        // name is always present but nullable
         const { id_number: idNumber, phone, date_of_birth: dob, address, name } = userInfo
 
-        let identity = null
+        // make cognito identities unique on:
+        // - phone number, or
+        // - combination of date of birth and name, or
+        // - id number
+        //    - if no id number, then we assume some sort of combination of phone, dob, address, and name
+        const identities = []
+        if (phone) {
+          identities.push(phone)
+        }
+        if (dob && name) {
+          identities.push(JSON.stringify({ dob, name }))
+        }
         if (idNumber) {
           const { value, category, type } = idNumber
-          identity = `${value}::${category}::${type}`
+          // reason why we did not JSON.stringify here is because
+          // there are already users whose masked identities were based on this format
+          identities.push(`${value}::${category}::${type}`)
         } else {
-          identity = JSON.stringify({ phone, dob, address, name })
+          // if webhook does not include id number, then we are expecting a few of the items below
+          identities.push(JSON.stringify({ phone, dob, address, name }))
```

Comment:
nice, makes sense to keep this one around for backwards compat, even though it's less sensitive than the phone / dob + name